pkg/user/signer_test.go

```diff
@@ -74,3 +75,43 @@ func (s *SignerTestSuite) ConfirmTxTimeout() {
 	require.Error(s.T(), err)
 	require.Equal(s.T(), err, context.DeadlineExceeded)
 }
+
+// TestGasConsumption verifies that the amount deducted from a user's balance is
+// based on the fee provided in the tx instead of the gas used by the tx. This
+// behavior leads to poor UX because tx submitters must over-estimate the amount
+// of gas that their tx will consume and they are not refunded for the excess.
+func (s *SignerTestSuite) TestGasConsumption() {
+	t := s.T()
+
+	utiaToSend := int64(1)
```

Comment:
[Optional] This test already has a fantastic set of comments 👍. Adding one more line to explain that this transaction specifies a fee amount of ` 1 tia`, which is higher than the actual gas consumption for this transaction, would elevate the explanation to the next level.

Reply:
Good point, can do!